### PR TITLE
Update fth

### PR DESCRIPTION
**Pull request by ruifanp**
# Description

Added resample_by option to fixed time horizon. Rewrote all unit tests and edited docs.

Fixes # (issue)

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

@Jackal08

### CODE REVIEW

File: docs/source/implementations/labeling_fixed_time_horizon.rst [hunk 1]
```diff
@@ -64,17 +76,26 @@ Below is an example on how to use the Fixed Horizon labeling technique on real d
...
+    labels = fixed_time_horizon(prices=data, threshold=1, lag=True, standardized=True, window=5)
 
-    # Create labels with standardization
-    labels = fixed_time_horizon(close=data[ticker], threshold=1, look_forward=1, standardized=True, window=5)
+    # Create labels after resampling weekly with standardization.
+    labels = fixed_time_horizon(prices=data, threshold=1, resample_by='W', lag=True,
+                                standardized=True, window=4)
 
 
 Research Notebook
```
Comment (PanPip) [inline docs/source/implementations/labeling_fixed_time_horizon.rst:101]: `Labeling` folder

File: mlfinlab/labeling/fixed_time_horizon.py [hunk 2]
```diff
@@ -1,67 +1,74 @@
...
+                        <https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects>`_
+    :param lag: (bool) If True, returns will be lagged to make them forward-looking.
     :param standardized: (bool) Whether returns are scaled by mean and standard deviation.
     :param window: (int) If standardized is True, the rolling window period for calculating the mean and standard
                     deviation of returns.
-    :return: (pd.Series) -1, 0, or 1 denoting whether return for each tick is under/between/greater than the threshold.
-                    The final look_forward number of observations will be labeled np.nan. Index is same as index of
-                    close.
+    :return: (pd.Series or pd.DataFrame) -1, 0, or 1 denoting whether the return for each observation is
+                    less/between/greater than the threshold at each corresponding time index.
```
Comment (PanPip) [inline mlfinlab/labeling/fixed_time_horizon.py:37]: Please, keep information that NaNs are present in the output.

File: mlfinlab/labeling/fixed_time_horizon.py [hunk 3]
```diff
@@ -1,67 +1,74 @@
...
 
-    # Warning if look_forward is greater than the length of the series.
-    if look_forward >= len(forward_return):
-        warnings.warn('look_forward period is greater than the length of the Series. All labels will be NaN.',
-                      UserWarning)
+    # Calculate returns.
+    if lag:
+        returns = prices.pct_change(1).shift(-1)
+    else:
+        returns = prices.pct_change(1)
```
Comment (PanPip) [inline mlfinlab/labeling/fixed_time_horizon.py:48]: Please, let the user define the period (for both simple and log returns).

File: mlfinlab/tests/test_labeling_fixed_time_horizon.py [hunk 4]
```diff
@@ -19,74 +19,114 @@ def setUp(self):
...
+        threshold = pd.Series([0.01]*10)  # No index
+        with self.assertRaises(Exception):
+            fixed_time_horizon(close, threshold)
         with self.assertRaises(Exception):
-            fixed_time_horizon(close, 0.01, look_forward=50, standardized=True)
+            fixed_time_horizon(close, 0.01, standardized=True)
         with self.assertWarns(UserWarning):
-            labels = fixed_time_horizon(close, 0.01, look_forward=50, standardized=True, window=50)
-        np.testing.assert_allclose(labels, [np.nan]*len(self.data['SPY'][:10]))
+            fixed_time_horizon(close, 0.01, standardized=True, window=50)
```
Comment (PanPip) [inline mlfinlab/tests/test_labeling_fixed_time_horizon.py:132]: Please, add commnets on which exceptions and warnings are you testing.

File: mlfinlab/labeling/fixed_time_horizon.py (hunk 3 above)
Comment (Jackal08) [reply mlfinlab/labeling/fixed_time_horizon.py:49]: See the reply in the excess median PR. Happy with this change.

**Comment by ruifanp** [comment]
@Jackal08 

ready to merge

**Pull request merged**